apps/deepsirius-ui/src/server/api/routers/ssh.ts

```diff
@@ -179,10 +203,16 @@ export const sshRouter = createTRPCRouter({
         },
       });
       if (!res.ok) {
-        throw new Error(`Failed to fetch: ${res.statusText}`);
+        const data: unknown = await res.json();
+        const error = apiErrorSchema.parse(data);
+
+        throw new TRPCError({
+          code: 'INTERNAL_SERVER_ERROR',
+          message: error.message,
+        });
       }
       const buffer = await res.arrayBuffer().catch((err: Error) => {
-        throw new Error(`Failed to fetch buffer file: ${err.message}`);
+        throw new Error(`Failed to fetch buffer file: ${err.message.replace(env.STORAGE_API_KEY, '***')}`);
```

Comment:
Not necessary here, the method `.arrayBuffer` does not throw the errors of the external API, i think. Maybe does not even need this catch..